cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java

```diff
@@ -547,26 +572,42 @@ protected List<EnvVar> getInitContainerEnvVars() {
      * It also generates the metrics and logging configuration. If this operand doesn't support logging
      * or metrics, they will not be set.
      *
-     * @param metricsAndLogging     The external CMs with logging and metrics configuration
-     *
+     * @param metricsAndLogging The external CMs with logging and metrics configuration
      * @return The generated ConfigMap
      */
     public ConfigMap generateBridgeConfigMap(MetricsAndLogging metricsAndLogging) {
         // generate the ConfigMap data entries for the metrics and logging configuration
         Map<String, String> data = ConfigMapUtils.generateMetricsAndLogConfigMapData(reconciliation, this, metricsAndLogging);
         // add the ConfigMap data entry for the bridge HTTP and Kafka clients related configuration
-        data.put(
-                BRIDGE_CONFIGURATION_FILENAME,
+        KafkaBridgeConfigurationBuilder builder =
                 new KafkaBridgeConfigurationBuilder(reconciliation, cluster, bootstrapServers)
                         .withTracing(tracing)
                         .withTls(tls)
                         .withAuthentication(authentication)
                         .withKafkaAdminClient(kafkaBridgeAdminClient)
                         .withKafkaProducer(kafkaBridgeProducer)
                         .withKafkaConsumer(kafkaBridgeConsumer)
-                        .withHttp(http, kafkaBridgeProducer, kafkaBridgeConsumer)
-                        .build()
-        );
+                        .withHttp(http, kafkaBridgeProducer, kafkaBridgeConsumer);
+
+        // mapping the old deprecated enableMetrics to JMX Exporter type
+        String metricsType;
+        if ((metrics instanceof JmxPrometheusExporterModel) || isMetricsEnabled) {
+            metricsType = JmxPrometheusExporterMetrics.TYPE_JMX_EXPORTER;
+        } else {
+            if (metrics instanceof StrimziMetricsReporterModel) {
+                metricsType = StrimziMetricsReporter.TYPE_STRIMZI_METRICS_REPORTER;
+            } else {
+                metricsType = "disabled";
+            }
+        }
+
+        if (metricsType.equals(JmxPrometheusExporterMetrics.TYPE_JMX_EXPORTER)) {
+            builder.withJmxPrometheusExporter((JmxPrometheusExporterModel) metrics, isMetricsEnabled);
+        } else if (metricsType.equals(StrimziMetricsReporter.TYPE_STRIMZI_METRICS_REPORTER)) {
+            builder.withStrimziMetricsReporter((StrimziMetricsReporterModel) metrics);
+        }
```

Comment:
Why exactly do we need two separate ifs here? Seems like it could be easily merged into one.

Reply:
Yep, sorry I didn't get what you mean. Done.